### ISSUE TITLE
The first run of patch.sh (3.8 branch) takes a very long time

### ISSUE THREAD

**Issue by bradfa** [beagleboard/kernel#22]
The first time you run patch.sh it clones Linus' entire tree, which on a slow connection, takes forever.  Could a `--depth 1` clone just of the tag of interest get the same state that's needed?  This would require that the tag of interest never goes backwards, which could be inconvenient.

**Comment by RobertCNelson** [comment]
If you have an existing tree, you can always pre-populate the kernel directory it with:

git clone --reference /folder/of/previous/git/clone git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git kernel

I use to do that all the time, back when my connection was very slow..

Just don't use a reference, that was previous cloned as refernce to, you'll eventually lose objects on git cleanup's..

Regards,

**Comment by bradfa** [comment]
@RobertCNelson Yes, but I'm thinking more of people who own Bones and want to try out the scripts.  Many of these people can follow exact steps but will get frustrated waiting for a 500+ MB git clone to take place, they also are unlikely to have a full git kernel tree laying around :).  If the default first run was a reasonable length due to a shallow clone, that would be helpful.  I'm working on a patch but hit some issues last night with it.  If I get it working I'll post a pull request.
